src/components/Status/Status.tsx

```diff
@@ -69,14 +69,16 @@ class Status extends UIComponent<Extendable<StatusProps>, any> {
     state: 'unknown',
   }
 
-  renderComponent({ ElementType, classes, rest, styles }) {
+  renderComponent({ ElementType, classes, rest, variables, styles }) {
     const { icon, renderIcon } = this.props as StatusPropsWithDefaults
     return (
       <ElementType {...rest} className={classes.root}>
         {Icon.create(icon, {
           defaultProps: {
             size: 'tiny',
-            variables: { color: 'white' }, // This is temporary. There is a ToDo to use icon's text/fill color for box-shadow, currently it uses color
+            styles: styles.icon,
+            variables: variables.icon,
+            className: classes.icon,
```

Comment:
these shouldn't be passed, as `classes.icon` contain already processed styles that are transformed to classes (`styles.icon` -> `classes.icon`). We should just pass original `styles.icon` and let `Icon` component process them. This is critical to ensure that these `styles.icon` will be correctly merged with those that could be provided in a following way:

```jsx
<Status icon={{ styles: ... }} />
```